include/cudawrappers/cu.hpp

```diff
@@ -668,6 +668,249 @@ class DeviceMemory : public Wrapper<CUdeviceptr> {
+      delete ptr;
+    });
+    cuCtxGetCurrent(&_context);
+  }
+
+  void addKernelNode(GraphNode &node,
+                     const std::vector<CUgraphNode> &dependencies,
+                     GraphKernelNodeParams &params) {
+    checkCudaCall(cuGraphAddKernelNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_KERNEL_NODE_PARAMS *>(&params)));
+  }
+
+  void addHostNode(GraphNode &node,
+                   const std::vector<CUgraphNode> &dependencies,
+                   GraphHostNodeParams &params) {
+    checkCudaCall(cuGraphAddHostNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_HOST_NODE_PARAMS *>(&params)));
+  }
+
+  void addDevMemFreeNode(GraphNode &node,
+                         const std::vector<CUgraphNode> &dependencies,
+                         const CUdeviceptr &devPtr) {
+    checkCudaCall(cuGraphAddMemFreeNode(node.getNode(), _obj,
+                                        dependencies.data(),
+                                        dependencies.size(), devPtr));
+  }
+
+  void addDevMemAllocNode(GraphNode &node,
+                          const std::vector<CUgraphNode> &dependencies,
+                          GraphDevMemAllocNodeParams &params) {
+    checkCudaCall(cuGraphAddMemAllocNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_MEM_ALLOC_NODE_PARAMS *>(&params)));
+  }
+
+  void addHostToDeviceMemCopyNode(GraphNode &node,
+                                  const std::vector<CUgraphNode> &dependencies,
+                                  GraphMemCopyToDeviceNodeParams &params) {
+#if defined(__HIP__)
+    hipMemcpy3DParms par_ = params;
+
+    checkCudaCall(hipGraphAddMemcpyNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(), &par_));
+
+#else
+    checkCudaCall(cuGraphAddMemcpyNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_MEMCPY3D *>(&params), _context));
+#endif
+  }
+
+  void addDeviceToHostMemCopyNode(GraphNode &node,
+                                  const std::vector<CUgraphNode> &dependencies,
+                                  GraphMemCopyToHostNodeParams &params) {
+#if defined(__HIP__)
+    hipMemcpy3DParms par_ = params;
+
+    checkCudaCall(hipGraphAddMemcpyNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(), &par_));
+
+#else
+    checkCudaCall(cuGraphAddMemcpyNode(
+        node.getNode(), _obj, dependencies.data(), dependencies.size(),
+        reinterpret_cast<CUDA_MEMCPY3D *>(&params), _context));
+#endif
+  }
+
+  void exportDotFile(
+      std::string path,
+      CUgraphDebugDot_flags flags =
+          CUgraphDebugDot_flags::CU_GRAPH_DEBUG_DOT_FLAGS_VERBOSE) {
+    checkCudaCall(cuGraphDebugDotPrint(_obj, path.c_str(), flags));
+  }
+
+  CUgraphExec Instantiate(unsigned int flags = CU_GRAPH_DEFAULT) {
```

Comment:
Why the capital I? Optional: rename to `instantiateWithFlags`. (I like the shorter `instantiate`, but for consistency it may be better to stick to NVIDIA's wording.
```suggestion
  CUgraphExec instantiate(unsigned int flags = CU_GRAPH_DEFAULT) {
```

Reply:
Changed to suggested naming